### PR TITLE
Support void transactions

### PR DESCRIPTION
**Pull request by vlaskhilkevich**
This is the rebased on latest changes version of PR #16

### CODE REVIEW

**Review by singh-sukhvir — COMMENTED** [review]
question around INVOICE_OPERATION property.

File: src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java [hunk 1]
```diff
@@ -83,6 +97,40 @@ public List<InvoiceItem> getAdditionalInvoiceItems(final Invoice invoice,
         }
     }
 
+    @Override
+    public OnSuccessInvoiceResult onSuccessCall(final InvoiceContext context, final Iterable<PluginProperty> properties) {
+        final String invoiceOperation = PluginProperties.findPluginPropertyValue(INVOICE_OPERATION, properties);
+        if (invoiceOperation == null) {
+            return super.onSuccessCall(context, properties);
+        }
```
[the remark this answers is not in the document]
Comment (KonradLubianka) [reply src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java:105]: it's property set internally by KillBill - please see this https://github.com/killbill/killbill/pull/1488/files

**Review by KonradLubianka — APPROVED** [review]

File: src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java [hunk 2]
```diff
@@ -83,6 +97,40 @@ public List<InvoiceItem> getAdditionalInvoiceItems(final Invoice invoice,
...
+                }
+            } catch (final ApiException e) {
+                logger.warn("Unable to {} transaction in Vertex", invoiceOperation, e);
+                // Don't fail the whole operation though
+            }
+        }
+
+        return super.onSuccessCall(context, properties);
+    }
+
```
Comment (singh-sukhvir) [inline src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java:133]: we need some tests for this implementation.

**Review by singh-sukhvir — CHANGES_REQUESTED** [review]
Need some tests. Looks good otherwise

File: src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java (hunk 2 above)
Comment (vlaskhilkevich) [reply src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java:136]: covered getAdditionalInvoiceItems and onSuccessCall in VertexInvoicePluginApiUnitTest

**Pull request merged**